### ISSUE TITLE
[Rust] Implement new Concept Exercise: string-use

### ISSUE THREAD

**Issue by efx** [exercism/rust#1060]
This issue describes how to implement the `string-use` concept exercise for the Rust track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how to work with `String`'s in Rust. This is part of a few concept exercises introducing the student to Rust's 2 string types.

## Learning objectives

- Know of the existence of type `String`.
- Know of basic functions that work on `String`.
- Strings are owned and thus can be mutable
- how print `String`'s using `format!` and basic formatting 

## Out of scope

- the difference between `String` and `&str` 
- performance characteristics (heap allocations, etc)

## Concepts

- strings-use

## Prerequisites

- intro-fn 
- intro-types 

### Hints

- [stdlib `String`](https://doc.rust-lang.org/std/string/index.html)
- ["Rust by Example" `String`](https://doc.rust-lang.org/stable/rust-by-example/std/str.html)
- [`std::fmt`](https://doc.rust-lang.org/std/fmt/index.html)

### After

- https://doc.rust-lang.org/book/ch08-02-strings.html#storing-utf-8-encoded-text-with-strings
- https://fasterthanli.me/blog/2020/working-with-strings-in-rust/

## Representer

Does not require changes to the Representer.

## Analyzer

Does not require an Analyzer.

## Implementing

See our [implementation guide](https://github.com/exercism/v3/blob/master/languages/rust/howto/implementing-a-concept-exercise.md#how-to-implement-a-rust-concept-exercise) for guidance.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.
